permissions/migrations/0015_unit_role_m2m_unit_and_groups.py

```diff
@@ -0,0 +1,32 @@
+# Generated by Django 3.2.12 on 2022-03-07 10:09
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```

Comment:
I am slightly concerned about this migration. This is a destructive operation and will remove all currently existing references to units and unit groups. Is this a problem?

If we want to preserve the existing units and unit groups, this migration must be written manually and I think it actually becomes a little tricky. Specially the reverse migration.

I can elaborate more if you do not catch what I am trying to say, but let's first establish if we do want to preserve data.

Reply:
Yup, good point - I'll write a custom migration

Reply:
And for backward reversible this is not. If this needs to be reversed we could do some manual handling for those.

Reply:
Yeah, to make it reversible it does require a decision of how to decide which unit/unit group should be preserved. Let's hope we do not have to reverse this migration. :sweat_smile:

Reply:
It actually could be done by creating multiple UnitRoles but still if we would unapply these it would be done manually or within some other migration that changes that back so its only logical to handle those cases then when know better.

Reply:
Yeah, you are right!

Changes good changes here. As this is not code that is constantly run in functions, I guess performance is not necessary. But, one thing to keep in mind. Since the `UnitRole` and `UnitGroup` is not used for grabbing data, I am fairly certain we do not actually have to fetch the objects from the database.

It would be enough to do this:

```pyton
role = UnitRole(id=role_id)
group = UnitGroup(id=unit_group_id)
role.unit_group.add(group)
```

Only the group update will hit the database. This is not necesary to change now.

Reply:
Didn't know that you could add the m2m relations like that (with instantiating only id to model).
Actually I had whole objects in memory at first but when tested it failed and needed to those instances to be refetched. But yeah have check that way out later :+1: